torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -310,6 +310,25 @@ def callback(fut):
     return out.result()
 
 
+def add_use_future_set_result(to, x, y, z):
+    out = rpc.Future()
+    fut = rpc.rpc_async(to, torch.add, args=(x, y))
+    fut.add_done_callback(lambda fut : out.set_result(fut.wait() + z))
+    return out.wait()
+
+
+def add_use_future_nested_cb(to, x, y, z):
+    out = rpc.Future()
+
+    def callback(fut1):
+        fut2 = rpc.rpc_async(to, torch.add, args=(fut1.wait(), z))
+        fut2.add_done_callback(lambda fut2 : out.set_result(fut2.wait()))
+
+    fut1 = rpc.rpc_async(to, torch.add, args=(x, y))
```

Comment:
Do we want to test calling UDFs that expect `rpc.Future` arguments?

Reply:
Good point, this is currently not a targeted use case. Let me customize `Future`'s pickle methods to through the correct error.